sdk/go/common/resource/plugin/provider_plugin.go

```diff
@@ -55,6 +55,10 @@ import (
 // Temporary feature flag to enable support for view resources.
 var supportsViews bool = cmdutil.IsTruthy(os.Getenv("PULUMI_ENABLE_VIEWS_PREVIEW"))
 
+// Temporary feature flag to enable support for provider-indicated refresh before update.
+var supportsProviderRefreshBeforeUpdate bool = cmdutil.IsTruthy(
+	os.Getenv("PULUMI_ENABLE_PROVIDER_REFRESH_BEFORE_UPDATE"))
+
```

Comment:
Do we need it? The protocol is pretty simple for this -- I don't think we'll adjust it based on trial/usage elsewhere, so we can probably merge/ship this change enabled by default without any envvar feature flag.

Reply:
Chatted with Will about this. Switched the envvar to be a "disable" flag, and we'll keep it temporarily in case it's useful to disable as part of testing with other providers. Opened a TODO issue (https://github.com/pulumi/pulumi/issues/19620) to remove the envvar in the near future.